### ISSUE TITLE
[TKLDev] Auto assign nameservers for build env (from TKLDev)

### ISSUE THREAD

**Issue by JedMeister** [turnkeylinux/tracker#388]
I have recently run into the issue of statically provided nameservers in the TKLDev build env. 

Generally the default works, but it would be much better if the build env pulled the DNS info from the TKLDev host. The recent issue I had was where TKLDev was given DNS info via DHCP and nameservers other than the local one were being blocked. 

The info provided in #95 assited me to resolve it but IMO if we want others to help out with TKL development and testing then we need to remove these sort of issues...

I have made some scripts which appear to work which I will provide soonish...
##

**Comment by l-arnold** [comment]
A bit complicated.  I have seen some scripts calling Google Nameservers in the TKL environment.  Guessing you might run a test:  IF Google DNS is found, then Use, otherwise find real DNS and use that.  ??

My nameservers are normally the same DNS structure even though IP may jump between a few zones.  Issues wtih DNS I have seen most often happen when built into Active Server as Active Server also wants the Windows Domain to have an internal (somewhat corrupted) DNS Server of its own that also references the Windows Domain.  (Samba doesn't really go there)

Separate subject really next and not so much for downloads, but for portability of the APP, I found that using "SERVER_ADDR" instead of "IPADDR" let a index.php page work referencing nearby ports by trimming the PORT number that gets associated with IPADDR.  (called on Odoo index.php from port 12325)

**Comment by l-arnold** [comment]
Is this error related to NameServers?  Has happened twice now.  I can navigate there from my workstation.  I saw something about Google Name Servers along the build process.  My Eth0 is using Comcast.  I can seem to find the missing file from my own workstation in:
http://archive.turnkeylinux.org/debian/pool/jessie/main/w/webmin-software/
here is the error:

* Bitvise xterm - 192.168.1.123:22
n x
Get:407 http://archive.turnkeylinux.org/debian/ jessie/main casper amd64 1.236-t
urnkey+12+gc47bd84 [19.3 kB]
Fetched 134 MB in 2min 35s <864 kB/s>
E: Failed to fetch http://archive.turnkeylinux.org/debian/pool/jessie/main/ii/web
min-shell/webmin-shell_l.740-turnkey+l+gl0debc0_all.deb 504 Host archive.turnk
eylinux.org lookup failed: Timeout
E: Failed to fetch http://archive.turnkeylinux.Org/debian/pool/jessie/main/w/web
min-software/webmin-software_l.740-turnkey+l+gl0debc0_all.deb 504 Host pi.turn
keylinux.org lookup failed: Timeout
E: Unable to fetch some archives, maybe run apt-get update or try with —fix-mis
traceback <most recent call last>

**Comment by l-arnold** [comment]
Looks like the issue is a space that has entered in.  Read Next (near the end of the error)

Host pi.turn keylinux.org lookup

**Comment by l-arnold** [comment]
Is this still an issue actually?  Seemed to work better and better last summer.

**Comment by JedMeister** [comment]
So long as you have access to Google public DNS (8.8.8.8) then this isn't a practical issue. However IMO we shouldn't hardcode google DNS IP into TKLDev. IMO it should automatically inherit the DNS servers of the TKLDev instance itself...

**Comment by l-arnold** [comment]
I agree.

**Comment by JedMeister** [comment]
Ok, I've made a start on this. PR can be viewed here: 

I'm not going to merge it yet though as it's not yet complete. I did a grep through `/usr` and it appears that `ruby` also has `8.8.8.8` hardcoded by the looks?!:
````
root@tkldev ~# grep -Fr "8.8.8.8" /usr
/usr/lib/ruby/2.1.0/resolv.rb:    # - :nameserver => '8.8.8.8'
/usr/lib/ruby/2.1.0/resolv.rb:    # - :nameserver => ['8.8.8.8', '8.8.4.4']
/usr/lib/ruby/2.1.0/resolv.rb:    # - :nameserver_port => [['8.8.8.8', 53], ['8.8.4.4', 53]]
````

**Comment by JedMeister** [comment]
Plus there is also the hardcoded nameservers in [common/mk/turnkey.mk](https://github.com/turnkeylinux/common/blob/master/mk/turnkey.mk#L36-L37).

I think that the best way to address this would be to have a `bash.rd.d` script which checks `/etc/resolv.conf` for nameservers and use those instead, but fallback to Google nameservers.

The above mentioned PR for TKLDev is probably part of the solution.

**Comment by JedMeister** [comment]
Closing this as it's been incidentally addressed for v17.0 via #1560 - Squid automagically uses the host's nameservers.
